scipy/linalg/special_matrices.py

```diff
@@ -1194,3 +1194,66 @@ def fiedler_companion(a):
     c[[0, 1], 0] = [-a[1], 1]
 
     return c
+
+def convmtx(a, n, mode='full'):
+    """
+    Construct a convolution matrix.
+
+    Constructs a matrix
+    `A = convmtx(a, n[, mode])`
+    such that
+    `np.dot(A, v)` is equivalent to `convolve(a, v[, mode])` but slower.
+
+    Parameters
+    ----------
+    a : The array to convolve.
+    n : The number of columns in the resulting matrix.
+        This is analogous to the length of v in numpy.convolve(v, a)
+    mode : This is analogous to `mode` in numpy.convolve(v, a, mode).
+        It determines the number of rows in A is
+            'full'[default]: `len(a) + n - 1`
+            'same': `max(n, len(a) )`
+            'valid': `max(n, len(a) ) - min(n, len(a) ) + 1`
+
+    Returns
+    -------
+    A : The convolution matrix
+
+    Notes
+    -----
+    .. versionadded:: 1.4.0
+
+    See Also
+    --------
+    toeplitz : Toeplitz matrix
+
+    Examples
+    --------
+    >>> from scipy.linalg import convmtx
+    >>> convmtx( (-1,2,-1), 5, mode='same')
+    array([[ 2, -1,  0,  0,  0],
+           [-1,  2, -1,  0,  0],
+           [ 0, -1,  2, -1,  0],
+           [ 0,  0, -1,  2, -1],
+           [ 0,  0,  0, -1,  2]])
+
+    """
+    a = np.asarray(a)
+    if a.ndim != 1:
+        raise ValueError('convmtx expects a 1d array as input')
```

Comment:
the line coverage looks pretty good for the new function, but I did notice that Azure reports no test hitting this exception handling